Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/cparser/CParserTask.java

```diff
@@ -59,85 +58,62 @@ public CParserTask(CParserPlugin plugin, String[] filenames, String options,
 
 	@Override
 	public void run(TaskMonitor monitor) {
-		DataTypeManager fileDtMgr = null;
 		try {
-			if (dtMgr == null) {
+			if (dataFileName != null) {
 				File file = new File(dataFileName);
 				dtMgr = FileDataTypeManager.createFileArchive(file);
-				fileDtMgr = dtMgr;
 			}
 
 			plugin.parse(filenames, options, dtMgr, monitor);
+			if (monitor.isCancelled()) {
+				SwingUtilities.invokeLater(() -> {
+					Msg.showInfo(getClass(),
+						plugin.getDialog().getComponent(), "Cancelled", "Task was cancelled.");
+				});
+			}
+
 			if (dataFileName != null) {
 				if (dtMgr.getDataTypeCount(true) != 0) {
 					try {
 						((FileDataTypeManager) dtMgr).save();
-						dtMgr.close();
-						SwingUtilities.invokeLater(new Runnable() {
-							@Override
-							public void run() {
-								Msg.showInfo(
-									getClass(), plugin.getDialog().getComponent(),
-									"Created Archive File", "Successfully created archive file\n" +
-										((FileDataTypeManager) dtMgr).getFilename());
-							}
-
+						SwingUtilities.invokeLater(() -> {
+							Msg.showInfo(getClass(),
+								plugin.getDialog().getComponent(),
+								"Created Archive File", "Successfully created archive file\n" +
+									((FileDataTypeManager) dtMgr).getFilename());
 						});
 					}
-					catch (DuplicateFileException e) {
-						Msg.showError(this, plugin.getDialog().getComponent(), "Error During Save",
-							e.getMessage());
-					}
 					catch (Exception e) {
 						Msg.showError(this, plugin.getDialog().getComponent(), "Error During Save",
 							"Could not save to file " + dataFileName, e);
 					}
 					finally {
-						if (dtMgr instanceof FileDataTypeManager) {
-							dtMgr.close();
-						}
+						dtMgr.close();
```

Comment:
https://github.com/NationalSecurityAgency/ghidra/blob/eaf6ab250df63652cd455f0c051ce7e03f4f641b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/database/data/ProgramDataTypeManager.java#L253-L256

Just in case there are any concerns over that change. :thumbsup: